Add custom registration success screen (#54)
pxblue/react-native-workflows: login-workflow/docs/customization.md

You can see a sample implementation of the custom details forms in the `/example` project.

> **NOTE:** If you are using a useEffect hook to call the `onDetailsChanged` function, you must make sure NOT to include the `onDetailsChanged` prop in your list of dependencies. This will cause an infinite update loop.

### Custom Registration Success Screen

You can customize the success screen shown at the end of the Registration flows using the `registrationSuccessScreen` prop on the `AuthUIContextProvider`. This prop gives you access to the `navigation` object so you can direct the user to the login page or elsewhere in your application after finishing registration as well as a `registrationData` object that contains a user's `AccountDetailInformation` and email:

```tsx
registrationSuccessScreen={(navigation, registrationData) => <MySuccessScreen navigation={navigation} email={registrationData.email}/>
```

> ### Custom Account Already Exists Success Screen
> 
> You can customize the success screen shown at the end of the Registration flows for users that already have an account using the `accountAlreadyExistsScreen` prop on the `AuthUIContextProvider`. This prop gives you access to the `navigation` object so you can direct the user to the login page or elsewhere in your application after finishing registration as well as a `registrationData` object that contains a user's `AccountDetailInformation` and email:
> 
> ```tsx
> accountAlreadyExistsScreen={(navigation, registrationData) => <MyAccountAlreadyExistsScreen navigation={navigation} email={registrationData.email} />
> ```

Reviewer: I would do this as a sub-section of the previous section.
```suggestion
#### Account Already Exists

In the case when a user already has an existing account, a separate success screen is shown. You can customize this screen in the same way using the `accountAlreadyExistsScreen` prop on the `AuthUIContextProvider`. This prop gives you access to the `navigation` object so you can direct the user to the login page or elsewhere in your application after finishing registration.:

```tsx
accountAlreadyExistsScreen={(navigation) => <MyAccountAlreadyExistsScreen navigation={navigation} />
```
```

You can see a sample implementation of the custom details forms in the `/example` project.

> **NOTE:** If you are using a useEffect hook to call the `onDetailsChanged` function, you must make sure NOT to include the `onDetailsChanged` prop in your list of dependencies. This will cause an infinite update loop.

### Custom Success Screen

> You can customize the success screen in the Registration flows by passing any `JSX.Element` to the `AuthUIContextProvider` via the `customRegistrationSuccessScreen` prop.

Reviewer: For this one you'll also want to mention that you can use the function too if you want access to the navigation object to go to other pages (pretty much everyone is going to need that or they'll be stuck on the success page forever mwahaha — once we take out the closeHeader).

> You can customize the success screen in the Registration flows by passing any `JSX.Element` to the `AuthUIContextProvider` via the `customRegistrationSuccessScreen` prop.

Reviewer: ```suggestion
You can customize the success screen shown at the end of the Registration flows using the `customRegistrationSuccessScreen` prop on the `AuthUIContextProvider`. This prop gives you access to the `navigation` object so you can direct the user to the login page or elsewhere in your application after finishing registration:
```
```
customRegistrationSuccessScreen={(navigation) => <MyCoolSuccessScreen navigation={navigation} />
```
